src/components/editor/codeMirror.ts

```diff
@@ -173,12 +172,45 @@ export function createCodeMirrorEditor(element: HTMLElement, cell: Cell, opts: {
 
     const readOnlyExtension = tagExtension('readOnly', EditorView.editable.of(!cell.metadata.properties.locked));
 
+    const cellSwitchExtension = keymap.of([
+        { key: "ArrowUp", run: (target) => { 
+                if(target.state.selection.ranges.length === 1 && target.state.selection.ranges[0].empty) {
+                    let firstLine = target.state.doc.line(1);
+                    let cursorPosition = target.state.selection.ranges[0].head;
+                    if(firstLine.from <= cursorPosition && cursorPosition <= firstLine.to) {
+                        _runtime.controls.emit({id: cell.id, type: "RUN_CELL", focusNext: "before"});
+                        return true
+                    } else {
+                        return false;
+                    }
+                } else {
+                    return false; 
+                }
+            }
+        },
+        { key: "ArrowDown", run: (target) => { 
+                if(target.state.selection.ranges.length == 1 && target.state.selection.ranges[0].empty) {
+                    let lastline = target.state.doc.line(target.state.doc.lines);
+                    let cursorPosition = target.state.selection.ranges[0].head;
+                    if(lastline.from <= cursorPosition && cursorPosition <= lastline.to) {
+                        _runtime.controls.emit({id: cell.id, type: "RUN_CELL", focusNext: "after"});
```

Comment:
Wait, do we actually want to `run` the cell or just focus the next? 
We can introduce a new message type `FOCUS_CELL` or something to just shift focus.


Also the `focusEditor` could take an optional argument:

```typescript
focusEditor(opts?: {at?: "start" | "end} = {}) {
  //...
}
 ```
But that can of course be future work outside of this PR :)

Reply:
That's a good point, I haven't considered that. I'm still purely thinking about reactive notebooks xD
Fixed it :)